mkosi

```diff
@@ -2417,16 +2423,56 @@ def install_arch(args: CommandLineArguments, root: str, do_run_build_script: boo
                     )
                 )
 
+        if args.bios_partno is not None:
+            vmlinuz_add_hook = os.path.join(hooks_dir, "90-mkosi-vmlinuz-add.hook")
+            with open(vmlinuz_add_hook, 'w') as f:
+                f.write(
+                    """\
+                    [Trigger]
+                    Operation = Install
+                    Operation = Upgrade
+                    Type = Path
+                    Target = usr/lib/modules/*/vmlinuz
+
+                    [Action]
+                    Description = Adding vmlinuz to /boot...
+                    When = PostTransaction
+                    Exec = /bin/bash -c 'while read -r f; do install -Dm644 "$f" /boot/vmlinuz-$(basename $(dirname "$f")); done'
```

Comment:
```suggestion
                    Exec = /bin/bash -c 'while read -r f; do install -Dm644 "$f" "/boot/vmlinuz-$(basename "$(dirname "$f")")"; done'
```
In the case of `f` containing spaces this would be the full quoting to be safe.